SYCL/ESIMD/dpas/dpas_fp16.cpp

```diff
@@ -20,14 +20,14 @@ int main(int argc, const char *argv[]) {
   bool Print = argc > 1 && std::string(argv[1]) == "-debug";
   bool Passed = true;
 
-  // Test unsigned 2-bit integers./
-  Passed &= tests<8, 8, fp16, fp16>(Q, Print);
-  Passed &= tests<8, 4, fp16, fp16>(Q, Print);
-  Passed &= tests<8, 1, fp16, fp16>(Q, Print);
+  constexpr bool LetDeduceArgs = true;
+  Passed &= tests<8, 8, fp16, fp16, LetDeduceArgs>(Q, Print);
+  Passed &= tests<8, 4, fp16, fp16, LetDeduceArgs>(Q, Print);
+  Passed &= tests<8, 1, fp16, fp16, LetDeduceArgs>(Q, Print);
 
   // TODO: Enable these cases when esimd::simd(ptr) constructor is fixed.
-  // Passed &= tests<8, 5, fp16, fp16>(Q, Print);
-  // Passed &= tests<8, 3, fp16, fp16>(Q, Print);
+  // Passed &= tests<8, 5, fp16, fp16, LetDeduceArgs>(Q, Print);
```

Comment:
as we talked, these testcases should be re-visited and maybe uncommented in a follow-up patch.